R/unused_import_linter.R

```diff
@@ -0,0 +1,60 @@
+#' Check that imported packages are actually used
+#'
+#' @param except_packages Character vector of packages that are ignored.
+#' These are usually attached for their side effects.
+#'
+#' @evalRd rd_tags("unused_import_linter")
+#' @seealso [linters] for a complete list of linters available in lintr.
+#' @export
+unused_import_linter <- function(except_packages = c("bit64", "data.table", "tidyverse")) {
+  Linter(function(source_file) {
+    if (is.null(source_file$full_xml_parsed_content)) return(list())
+
+    import_exprs <- xml2::xml_find_all(
+      source_file$full_xml_parsed_content,
+      paste(
+        "//expr[",
+        "expr[SYMBOL_FUNCTION_CALL[text() = 'library' or text() = 'require']]",
+        "and",
+        "not(SYMBOL_SUB[text() = 'character.only'])",
+        "]"
+      )
+    )
+    if (length(import_exprs) == 0) {
+      return(list())
+    }
+    # strip_names() needed to turn "'pkg'" -> "pkg" for STR_CONST calls.
```

Comment:
I don't think we want `strip_names()` here, right? e.g. for `R"()"` strings

Reply:
Yes, it's a quick-and-dirty solution. Should be revisited when we unify the string literal treatment.
Do we already have an issue for that? Then I'll mark it as a TODO for that issue.

Reply:
true, but I had in mind consistency with missing_package_linter, which is in turn consistent with what library itself does.

Reply:
#1034 is the issue btw

Reply:
ok, will change to that. This approach also copes with raw strings, so no follow-up todo 🥳